### PR TITLE
Fix message bar being drawn over terminal content after the next resize

### LINKED ISSUE #2924: Message bar is overlapping terminal content
Introduced in #2438. 

so instead of
```
-----------------------
|                     |
|       terminal      |
|                     |
----------------------- <- terminal end
----------------------- <- message bar start
|     message bar     |
----------------------- <- message bar end
```
we now have
```
-----------------------
|                     |
|       terminal      |
|                     |
----------------------- <- message bar start
|     message bar     |
----------------------- <- message bar and terminal end
```

### PR DESCRIPTION
**Pull request by kchibisov**
Fixes #2924

### CODE REVIEW

File: alacritty/src/event.rs [hunk 1]
```diff
@@ -488,8 +488,15 @@ impl<N: Notify> Processor<N> {
                     CloseRequested => processor.ctx.terminal.exit(),
                     Resized(lsize) => {
                         let psize = lsize.to_physical(processor.ctx.size_info.dpr);
-                        processor.ctx.resize_pending.dimensions = Some(psize);
-                        processor.ctx.terminal.dirty = true;
+                        // Drop resize events, which resize to the current size
+                        if (psize.height as f32 - processor.ctx.size_info.height).abs()
```
Comment (kchibisov) [inline alacritty/src/event.rs]: This change is an optimization for resizes, that are identical to a current window size. If you don't like it, I'm fine with ignoring it.

Comment (chrisduerr) [reply alacritty/src/event.rs]: When is this ever hit? Why would a resize event be sent in that case?

Comment (kchibisov) [reply alacritty/src/event.rs]: Change focus on sway. So if you have two windows next to each other and you change a focus, you'll get a resize.

Comment (chrisduerr) [reply alacritty/src/event.rs]: That sounds like a winit bug to me?

Comment (kchibisov) [reply alacritty/src/event.rs]: Oh, it does actually. Doesn't happen on `X11`. Should I revert this `resize` check?

Comment (chrisduerr) [reply alacritty/src/event.rs]: Yeah and report upstream probably. We shouldn't be checking if resize events make sense, that's something winit should be doing.

Comment (kchibisov) [reply alacritty/src/event.rs]: https://github.com/rust-windowing/winit/issues/1247

**Pull request merged**